chain/src/txhashset/txhashset.rs

```diff
@@ -115,12 +114,12 @@ impl TxHashSet {
 		root_dir: String,
 		commit_index: Arc<ChainStore>,
 		header: Option<&BlockHeader>,
-		sandbox: bool,
+		sandbox_dir: Option<String>,
```

Comment:
We can just pass _either_ `root_dir` _or_ `sandbox_dir` as the first param and we don't need to handle the precedence rule here.

Reply:
Thanks @antiochp for the review.

You have `header/header_head` and `header/sync_head/` reading on this `open()` function,  but `header/` is at same level as `txhashset/` so surely it doesn't exist in sandbox folder.

That's why I keep this first `root_dir` param. 

Could you explain more about why we need put `header\` also into the `sandbox` folder?

```
			header_pmmr_h: PMMRHandle::new(
				&root_dir,
				HEADERHASHSET_SUBDIR,
				HEADER_HEAD_SUBDIR,
				false,
				None,
			)?,
			sync_pmmr_h: PMMRHandle::new(
				&root_dir,
				HEADERHASHSET_SUBDIR,
				SYNC_HEAD_SUBDIR,
				false,
				None,
			)?,
```

Reply:
Ah, perhaps that `rebuild` did the job :-)
```
		self.rebuild_header_mmr(&Tip::from_header(&header), &mut txhashset)?;
```

Reply:
but then what's the best to handle this `header/` folder in `sandbox`?
1. also move to the db root
2. call a `rebuild_header_mmr()` again after moving the `txhashset/` folder

Which one do you prefer?  Is the `rebuild_header_mmr()` a fast or slow processing?

Reply:
Tried with `moving the header/ folder`, but see a lot of `Corrupted storage` log which could be caused by not-atomic processing on store during `txhashset_write` processing?

Switch to `rebuild_header_mmr()` solution now.
And regarding the speed, the `rebuilding` spent 10 seconds on my slow mac air for current main chain height. not bad.

Reply:
Ahh ok.

```
chain_data/txhashset <- txhashset.zip unzips to here
chain_data/header <- local data only, not included in the zip file
```

The original intent was to keep the `header` dir out the `txhashset` dir to make it more explicit what originated from `txhashset.zip` and what was purely built locally.

Maybe we consider making the sandbox dir a couple of levels deep?

So we can have something like - 

```
chain_data/txhashset
chain_data/header
```

```
$sandbox_dir/txhashset
$sandbox_dir/header
```

Would this work?

I don't see we would not be able to copy across both the `txhashset` dir _and_ the `header` dir on successful validation.
A txhashset will have file references to files under both these dirs so we'd need to be careful with ordering.
i.e. copy them across then reopen the newly updated txhashset.

Reply:
@antiochp I don't know why, I tried the `header` dir moving on https://github.com/mimblewimble/grin/pull/2685/commits/2a6182b1a6d401fbde72457d1a8327c48b88c319, but there were a lot of `Corrupted storage` logs. So I switch to `rebuild_header_mmr` solution.